drivers/sensor/i3g4250d/i3g4250d.c

```diff
@@ -0,0 +1,230 @@
+/* ST Microelectronics I3G4250D gyro driver
+ *
+ * Copyright (c) 2021 Jonathan Hahn
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Datasheet:
+ * https://www.st.com/resource/en/datasheet/i3g4250d.pdf
+ */
+
+#define DT_DRV_COMPAT st_i3g4250d
+
+#include <init.h>
+#include <sys/__assert.h>
+#include <sys/byteorder.h>
+#include <drivers/sensor.h>
+#include <logging/log.h>
+
+#include "i3g4250d.h"
+
+#define RAW_TO_MICRODEGREEPERSEC 8750
+
+LOG_MODULE_REGISTER(i3g4250d, CONFIG_SENSOR_LOG_LEVEL);
+
+static int i3g4250d_sample_fetch(const struct device *dev,
+			enum sensor_channel chan)
+{
+
+	struct i3g4250d_data *i3g4250d = dev->data;
+	int ret;
+	uint8_t reg;
+	int16_t buf[3] = { 0 };
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL || chan == SENSOR_CHAN_GYRO_XYZ);
+
+	ret = i3g4250d_flag_data_ready_get(i3g4250d->ctx, &reg);
+
+	if (ret < 0 || reg != 1) {
+		return ret;
+	}
+
+	ret = i3g4250d_angular_rate_raw_get(i3g4250d->ctx, buf);
+	if (ret < 0) {
+		LOG_ERR("Failed to fetch raw data sample!");
+		return ret;
+	}
+
+	memcpy(i3g4250d->angular_rate, buf, sizeof(i3g4250d->angular_rate));
```

Comment:
Need to handle byte order?

Reply:
Looking at the [implementation](https://github.com/zephyrproject-rtos/hal_st/blob/575de9d461aa6f430cf62c58a053675377e700f3/sensor/stmemsc/i3g4250d_STdC/driver/i3g4250d_reg.c#L338), I would say not needed. Angular rate values are processed byte wise and the resulting 16bit int is calculated by adding low byte to high byte multiplied by 256, which should work independent of the byte order of the mcu or did I miss something?

Reply:
After looking at the implementation, I agree swapping isn't needed here.